storybook-addon-export-to-codesandbox/src/withCodeSandboxButton.ts

```diff
@@ -0,0 +1,139 @@
+import { StoryFn as StoryFunction, StoryContext, useEffect, StoryWrapper } from '@storybook/addons';
+import { getParameters } from 'codesandbox-import-utils/lib/api/define';
+import dedent from 'dedent';
+
+const dependencyRegex = / from '.*?'; \/\/ codesandbox-dependency: (.*?) (.*)/g;
+const dependencySubs = " from '$1';";
+
+export const withCodeSandboxButton: StoryWrapper = (StoryFn: StoryFunction, context: StoryContext) => {
+  if (context.viewMode === 'docs') {
+    useEffect(() => {
+      displayToolState(`#anchor--${context.id} .docs-story`, context);
+    });
+  }
+
+  return StoryFn(context);
+};
+
+const getDependencies = (fileContent: string, dependencies: { [dependencyName: string]: string }) => {
```

Comment:
```suggestion
const getDependencies = (fileContent: string, requiredDependencies: { [dependencyName: string]: string }) => {
```

_nit:_ let's use the same variable name as in its call

---

This change will highlight that this function is not pure as it mutates its input, let's clone its input to make it safer:

```tsx
const getDependencies = (fileContent: string, requiredDependencies: { [dependencyName: string]: string }) => {
  const dependencies = { ...requiredDependencies };
```